cub/cub/device/device_segmented_reduce.cuh

```diff
@@ -73,6 +78,66 @@ CUB_NAMESPACE_BEGIN
 //! @endrst
 struct DeviceSegmentedReduce
 {
+private:
+  template <typename InputIteratorT,
+            typename OutputIteratorT,
+            typename BeginOffsetIteratorT,
+            typename EndOffsetIteratorT,
+            typename OffsetT,
+            typename ReductionOpT,
+            typename InitT>
+  CUB_RUNTIME_FUNCTION static cudaError_t custom_segmented_reduce(
+    ::cuda::std::false_type,
+    void* d_temp_storage,
+    std::size_t& temp_storage_bytes,
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    int num_segments,
+    BeginOffsetIteratorT d_begin_offsets,
+    EndOffsetIteratorT d_end_offsets,
+    ReductionOpT reduction_op,
+    InitT initial_value,
+    cudaStream_t stream);
+
+  template <typename InputIteratorT,
+            typename OutputIteratorT,
+            typename BeginOffsetIteratorT,
+            typename EndOffsetIteratorT,
+            typename OffsetT,
+            typename ReductionOpT,
+            typename InitT>
+  CUB_RUNTIME_FUNCTION static cudaError_t custom_segmented_reduce(
+    ::cuda::std::true_type,
```

Comment:
I believe that one can go. By adding the internal static assert we are already prohibiting any instantiations of `custom_segmented_reduce` where this is `false_type`

Please consider removing that extra function and just using the static_assert